src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -117,15 +118,15 @@ uct_cuda_ipc_iface_is_reachable_v2(const uct_iface_h tl_iface,
     }
 
 #if HAVE_CUDA_FABRIC
-    if (uct_cuda_ipc_iface_is_mnnvl_supported(md)) {
-        /* multi-node nvlink is supported and enabled */
-        return 1;
-    }
+    mnnvl_supported = uct_cuda_ipc_iface_is_mnnvl_supported(md);
+#else
+    mnnvl_supported = 0;
 #endif
 
-    /* Not fabric capable or multi-node nvlink disabled, so iface has to be on
-     * the same node for cuda-ipc to be reachable */
-    if ((ucs_get_system_id() != *((const uint64_t*)params->device_addr))) {
+    /* Either multi-node NVLINK should be supported or iface has to be on the
+     * same node for cuda-ipc to be reachable */
+    if ((ucs_get_system_id() != *((const uint64_t*)params->device_addr)) &&
+        !mnnvl_supported) {
```

Comment:
can we call uct_cuda_ipc_iface_is_mnnvl_supported() here, and uct_cuda_ipc_iface_is_mnnvl_supported would be defined alway (return 0 if HAVE_CUDA_FABRIC is undefined)
this way, no need to query MNNVL if on same system, since we already know it is reachable

Reply:
and remove `#if` when calling uct_cuda_ipc_iface_is_mnnvl_supported

Reply:
done